tensorflow/core/kernels/maxpooling_op.cc

```diff
@@ -965,7 +975,9 @@ struct LaunchMaxPoolingGradWithArgmax<CPUDevice, T> {
         const int input_start = start * input_size_per_batch;
         const int input_end = limit * input_size_per_batch;
         for (int64 index = input_start; index < input_end; index++) {
-          const int64 grad_out_index = argmax_flat(index);
+          const int64 grad_out_index = include_batch_in_index
+                                           ? argmax_flat(index)
+                                           : argmax_flat(index) + output_start;
```

Comment:
If `include_batch_in_index` is false, `grad_out_index` won't necessarily be correct if this shard processes two or more batches. `output_start` indexes into the first batch, so adding `output_start` is not correct on the second batch.

The correct logic is something like:

```
int64 grad_out_index = argmax_flat(index);
if (!include_batch_in_index) {
  cur_batch = index / input_size_per_batch;
  grad_out_index += cur_batch * output_size_per_batch;
}
```